test/test_testing.py

```diff
@@ -4016,16 +4101,45 @@ def test_push_and_pull_non_utf8_data(self):
             encoding='sjis',
             stream_class=io.StringIO)
 
-    def _test_push_and_pull_data(self, original_data, encoding, stream_class):
+    if typing.TYPE_CHECKING:
+        @typing.overload
```

Comment:
I reckon we drop these overloads on this internal function. They don't seem to be necessary with the casts you have below.

Reply:
Ah, indeed. I added those first and then had to do the other changes anyway, and you're right that they are no longer required. Thanks!